### PR TITLE
Apple Silicon "MPS" support main examples

### PR DESCRIPTION
**Pull request by NripeshN**
Hi @rusty1s, The following files passed successfully on "MPS" device

### CODE REVIEW

**Comment by NripeshN** [comment]
Hi @rusty1s @wsad1,
I have tested all these examples on my m2 max MacBook Pro and they have successfully run.

File: examples/argva_node_clustering.py [hunk 1]
```diff
@@ -15,7 +15,12 @@
...
 from torch_geometric.nn import ARGVA, GCNConv
 
-device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+if torch.cuda.is_available():
+    device = torch.device('cuda')
+elif hasattr(torch.backends, 'mps') and torch.backends.mps.is_available():
+    device = torch.device('mps')
+else:
+    device = torch.device('cpu')
 transform = T.Compose([
```
Comment (rusty1s) [inline examples/argva_node_clustering.py:24]:
```suggestion
    device = torch.device('cpu')

transform = T.Compose([
```

File: examples/dna.py [hunk 2]
```diff
@@ -59,7 +59,12 @@ def forward(self, x, edge_index):
...
 
 
-device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+if torch.cuda.is_available():
+    device = torch.device('cuda')
+elif hasattr(torch.backends, 'mps') and torch.backends.mps.is_available():
+    device = torch.device('mps')
+else:
+    device = torch.device('cpu')
 model = Net(in_channels=dataset.num_features, hidden_channels=128,
```
Comment (rusty1s) [inline examples/dna.py:68]:
```suggestion
    device = torch.device('cpu')

model = Net(in_channels=dataset.num_features, hidden_channels=128,
```

File: examples/tagcn.py [hunk 3]
```diff
@@ -27,7 +27,12 @@ def forward(self):
         return F.log_softmax(x, dim=1)
 
 
-device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+if torch.cuda.is_available():
+    device = torch.device('cuda')
+elif hasattr(torch.backends, 'mps') and torch.backends.mps.is_available():
+    device = torch.device('mps')
+else:
+    device = torch.device('cpu')
```
Comment (rusty1s) [inline examples/tagcn.py:35]:
```suggestion
    device = torch.device('cpu')

```

**Review by rusty1s — APPROVED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -63,14 +63,15 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
...
 - Added instructions for ROCm build wheels ([#7143](https://github.com/pyg-team/pytorch_geometric/pull/7143))
 - Added a `ComposeFilters` class to compose `pre_filter` functions in `Dataset` ([#7097](https://github.com/pyg-team/pytorch_geometric/pull/7097))
 - Added a time-step aware variant of the `EllipticBitcoinDataset` called `EllipticBitcoinTemporalDataset` ([#7011](https://github.com/pyg-team/pytorch_geometric/pull/7011))
 - Added `to_dgl` and `from_dgl` conversion functions ([#7053](https://github.com/pyg-team/pytorch_geometric/pull/7053))
 - Added support for `torch.jit.script` within `MessagePassing` layers without `torch_sparse` being installed ([#7061](https://github.com/pyg-team/pytorch_geometric/pull/7061), [#7062](https://github.com/pyg-team/pytorch_geometric/pull/7062))
 - Added unbatching logic for `torch.sparse` tensors ([#7037](https://github.com/pyg-team/pytorch_geometric/pull/7037))
 - Added the `RotatE` KGE model ([#7026](https://github.com/pyg-team/pytorch_geometric/pull/7026))
+- Added support for apple silicon GPU Acceleration for some main examples  (**[**#**7770**]**(**https://github.com/pyg-**team**/pytorch_geometric/pull/7770**)**)
```
Comment (rusty1s) [inline CHANGELOG.md:74]:
```suggestion
- Added support for Apple silicon GPU acceleration in some main examples  ([#7770](https://github.com/pyg-team/pytorch_geometric/pull/7770))
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -63,14 +63,15 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
...
 - Added instructions for ROCm build wheels ([#7143](https://github.com/pyg-team/pytorch_geometric/pull/7143))
 - Added a `ComposeFilters` class to compose `pre_filter` functions in `Dataset` ([#7097](https://github.com/pyg-team/pytorch_geometric/pull/7097))
 - Added a time-step aware variant of the `EllipticBitcoinDataset` called `EllipticBitcoinTemporalDataset` ([#7011](https://github.com/pyg-team/pytorch_geometric/pull/7011))
 - Added `to_dgl` and `from_dgl` conversion functions ([#7053](https://github.com/pyg-team/pytorch_geometric/pull/7053))
 - Added support for `torch.jit.script` within `MessagePassing` layers without `torch_sparse` being installed ([#7061](https://github.com/pyg-team/pytorch_geometric/pull/7061), [#7062](https://github.com/pyg-team/pytorch_geometric/pull/7062))
 - Added unbatching logic for `torch.sparse` tensors ([#7037](https://github.com/pyg-team/pytorch_geometric/pull/7037))
 - Added the `RotatE` KGE model ([#7026](https://github.com/pyg-team/pytorch_geometric/pull/7026))
+- Added support for Apple silicon GPU acceleration in some main examples  ([#7770](https://github.com/pyg-team/pytorch_geometric/pull/7770))
```
Comment (rusty1s) [inline CHANGELOG.md:74]:
```suggestion
- Added support for Apple silicon GPU acceleration in some main examples ([#7770](https://github.com/pyg-team/pytorch_geometric/pull/7770))
```

**Pull request merged**